### ISSUE TITLE
[java]RedundantFieldInitializer can not detect a special case for char initialize: char foo = '\0';

### ISSUE THREAD

**Issue by wuchiuwong** [pmd/pmd#2441]
**Affects PMD Version:** 
6.22.0

**Rule:**
RedundantFieldInitializer

**Description:**
RedundantFieldInitializer can not detect a special case for <code>char</code>  initialize: <code>char foo = '\0';</code>
The reason for the undetectability is that <code>'\0'</code> can't convert to int directly, in <code>value = (int) literal.getImage().charAt(1);</code> in pmd-java-6.22.0-sources.jar!\net\sourceforge\pmd\lang\java\rule\performance\RedundantFieldInitializerRule.java_93
**Code Sample demonstrating the issue:**

```
char foo = '\0';
```

**Expected outcome:**
false-negative

**Running PMD through:**
CLI
